### ISSUE TITLE
Native TF and Backend TF performance comparison

### ISSUE THREAD

**Issue by emrahyigit** [fchollet/keras#5109]
Hey everyone, 

I created the exact same network with native and backend tensorflow but after many hours of testing using number of different parameters, still couldn't figure out why keras outperforms the native tensorflow, does better convergences and produces better(slightly but better) results. 

Does Keras implement a different weight initializer method? or performs different weight decay approach other than **tf.train.inverse_time_decay**? 

P.s. the score difference is always like 
Keras with Tensorflow: ~0.9850 - 0.9885
Tensorflow Native ~0.9780 - 0.9830 

Thank you in advance

**Comment by yaroslavvb** [comment]
@fchollet in case he feels like digging in, but these questions are generally better suited for StackOverflow. The general approach to answering these questions is to use timeline profiling from https://github.com/tensorflow/tensorflow/issues/1824#issuecomment-225754659 , going through all the ops that are getting called in each method, and then going through Keras code to find why those ops are getting created

**Comment by emrahyigit** [comment]
By the way, 

My environment is:

> Python 3.5.2 -Anaconda / Windows 10
> CUDA: 8.0 with cuDNN 5.1 
> Keras 1.2.1
> Tensorflow 0.12.1
> Nvidia Geforce GTX 860M

and **keras.json** file:

```
{
    "image_dim_ordering": "th", 
    "epsilon": 1e-07, 
    "floatx": "float32", 
    "backend": "tensorflow"
}
```



[Codes.zip](https://github.com/fchollet/keras/files/720997/Codes.zip)

**Comment by emrahyigit** [comment]
Shame... I forgot to add **bias** to first two convolutions. 

Sorry for inconvenience.

**Comment by emrahyigit** [comment]
I also tried that with Lasagne and it produced the same results as TensorFlow. 

What's going on? I am going to scream soon!
